### ISSUE TITLE
Add `Proxy.advise` (and maybe `Proxy.on`?)

### ISSUE THREAD

**Issue by briancavalier** [cujojs/wire#108]
If proxies provided an api for advising, then they could implement their own custom way of advising things that we can't currently advise with meld.  wire/aop could simply rely on `Proxy.advise`, and wire's base proxy could use meld to implement `Proxy.advise`.

For example, we could advise "non-standard" objects like jQuery widgets if the jQuery widget proxy implemented a custom `advise` method.

We could do something similar with `Proxy.on` for easier integration with other framework's custom event systems (for which _none_ of our existing facets will work).

In short, this may help us integrate other libraries and frameworks more easily without having to write custom facets.  It puts more burden on Proxy implementations, but allows facets (which are the cujoJS developers interface to wire) to be much more generic.
